### ISSUE TITLE
Segfault when disconnecting output

### ISSUE THREAD

**Issue by vilhalmer** [vilhalmer/oguri#4]
```
Jun 23 21:25:34 sayaka systemd-coredump[7883]: Process 603 (oguri) of user 1000 dumped core.
                                               
                                               Stack trace of thread 603:
                                               #0  0x00007fad2d03de14 wl_proxy_marshal (libwayland-client.so.0)
                                               #1  0x00005575ac1c577c n/a (/home/vil/Code/oguri/build/oguri)
                                               #2  0x00005575ac1c5d73 n/a (/home/vil/Code/oguri/build/oguri)
                                               #3  0x00005575ac1c6bf4 n/a (/home/vil/Code/oguri/build/oguri)
                                               #4  0x00007fad2cc9f06b __libc_start_main (libc.so.6)
                                               #5  0x00005575ac1c546a n/a (/home/vil/Code/oguri/build/oguri)
```

**Comment by emersion** [comment]
Cannot reproduce. Is it possible to get a more detailed stack trace?

**Comment by vilhalmer** [comment]
I think I fixed this in f93bc7f, I haven't seen it recently. It was likely just due to the program outliving the output object it was trying to use and some coincidence of timers.

**Comment by vilhalmer** [comment]
Closing, will reopen if I see it again.
